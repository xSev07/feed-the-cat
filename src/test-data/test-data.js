const products = [
  {
    id: `fg_small`,
    caption: `Сказочное заморское яство`,
    brand: `Нямушка`,
    taste: `c фуа-гра`,
    numberOfServings: 10,
    weight: 0.5,
    unit: `кг`,
    description: `Печень перекормленного гуся`,
    balance: 10,
  },
  {
    id: `fish_medium`,
    caption: `Сказочное заморское яство`,
    brand: `Нямушка`,
    taste: `c рыбой`,
    numberOfServings: 40,
    weight: 2,
    unit: `кг`,
    description: `Головы щучьи с чесноком да свежайшая сёмгушка.`,
    balance: 20,
  },
  {
    id: `chicken_large`,
    caption: `Сказочное заморское яство`,
    brand: `Нямушка`,
    taste: `c курой`,
    numberOfServings: 100,
    weight: 5,
    unit: `кг`,
    description: `Сочные ножки Буша`,
    balance: 0,
  },
];

export {products};
