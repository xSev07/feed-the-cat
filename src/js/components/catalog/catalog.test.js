import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./catalog";
import {products} from "../../test-data/test-data";

it(`Should Catalog render correctly`, () => {
  const tree = renderer
    .create(
        <Catalog
          products={products}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
