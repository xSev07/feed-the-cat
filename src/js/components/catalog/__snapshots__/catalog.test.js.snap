// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Catalog render correctly 1`] = `
<section
  className="catalog"
>
  <div
    className="catalog__container container"
  >
    <h2
      className="catalog__title"
    >
      Ты сегодня покормил кота?
    </h2>
    <form>
      <ul
        className="catalog__list"
      >
        <li
          className="catalog__item "
        >
          <input
            checked={false}
            className="hidden"
            id="fg_small"
            readOnly={true}
            type="checkbox"
          />
          <a
            className="catalog__content"
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <p
              className="catalog__caption-main"
            >
              Сказочное заморское яство
            </p>
            <h3
              className="catalog__name-container"
            >
              <span
                className="catalog__name-brand"
              >
                Нямушка
              </span>
              <br />
              <span
                className="catalog__name-taste"
              >
                c фуа-гра
              </span>
            </h3>
            <p
              className="catalog__spec"
            >
              <span
                className="catalog__quantity"
              >
                10
                 
              </span>
              порций
              <br />
              мышь
               в подарок
              <br />
            </p>
            <div
              className="catalog__picture-container"
            >
              <picture>
                <source
                  srcSet="assets/img/cat_preview.webp"
                  type="image/webp"
                />
                <img
                  alt="Нямушка c фуа-гра"
                  className="catalog__picture"
                  height="275"
                  src="assets/img/cat_preview.png"
                  width="260"
                />
              </picture>
            </div>
            <div
              className="catalog__size-container"
            >
              <p
                className="catalog__size-value"
              >
                0,5
              </p>
              <p
                className="catalog__size-unit"
              >
                кг
              </p>
            </div>
          </a>
          <p
            className="catalog__caption-additional"
          >
            Чего сидишь? Порадуй котэ,
            <a
              className="catalog__buy"
              href="#"
              onClick={[Function]}
            >
               купи
            </a>
          </p>
        </li>
        <li
          className="catalog__item "
        >
          <input
            checked={false}
            className="hidden"
            id="fish_medium"
            readOnly={true}
            type="checkbox"
          />
          <a
            className="catalog__content"
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <p
              className="catalog__caption-main"
            >
              Сказочное заморское яство
            </p>
            <h3
              className="catalog__name-container"
            >
              <span
                className="catalog__name-brand"
              >
                Нямушка
              </span>
              <br />
              <span
                className="catalog__name-taste"
              >
                c рыбой
              </span>
            </h3>
            <p
              className="catalog__spec"
            >
              <span
                className="catalog__quantity"
              >
                40
                 
              </span>
              порций
              <br />
              <span
                className="catalog__quantity"
              >
                2
                 
              </span>
              мыши
               в подарок
              <br />
            </p>
            <div
              className="catalog__picture-container"
            >
              <picture>
                <source
                  srcSet="assets/img/cat_preview.webp"
                  type="image/webp"
                />
                <img
                  alt="Нямушка c рыбой"
                  className="catalog__picture"
                  height="275"
                  src="assets/img/cat_preview.png"
                  width="260"
                />
              </picture>
            </div>
            <div
              className="catalog__size-container"
            >
              <p
                className="catalog__size-value"
              >
                2
              </p>
              <p
                className="catalog__size-unit"
              >
                кг
              </p>
            </div>
          </a>
          <p
            className="catalog__caption-additional"
          >
            Чего сидишь? Порадуй котэ,
            <a
              className="catalog__buy"
              href="#"
              onClick={[Function]}
            >
               купи
            </a>
          </p>
        </li>
        <li
          className="catalog__item catalog__item--disabled "
        >
          <input
            checked={false}
            className="hidden"
            id="chicken_large"
            readOnly={true}
            type="checkbox"
          />
          <a
            className="catalog__content"
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <p
              className="catalog__caption-main"
            >
              Сказочное заморское яство
            </p>
            <h3
              className="catalog__name-container"
            >
              <span
                className="catalog__name-brand"
              >
                Нямушка
              </span>
              <br />
              <span
                className="catalog__name-taste"
              >
                c курой
              </span>
            </h3>
            <p
              className="catalog__spec"
            >
              <span
                className="catalog__quantity"
              >
                100
                 
              </span>
              порций
              <br />
              <span
                className="catalog__quantity"
              >
                5
                 
              </span>
              мышей
               в подарок
              <br />
              заказчик доволен
            </p>
            <div
              className="catalog__picture-container"
            >
              <picture>
                <source
                  srcSet="assets/img/cat_preview.webp"
                  type="image/webp"
                />
                <img
                  alt="Нямушка c курой"
                  className="catalog__picture"
                  height="275"
                  src="assets/img/cat_preview.png"
                  width="260"
                />
              </picture>
            </div>
            <div
              className="catalog__size-container"
            >
              <p
                className="catalog__size-value"
              >
                5
              </p>
              <p
                className="catalog__size-unit"
              >
                кг
              </p>
            </div>
          </a>
          <p
            className="catalog__caption-additional"
          >
            Печалька, c курой закончился
          </p>
        </li>
      </ul>
    </form>
  </div>
</section>
`;
